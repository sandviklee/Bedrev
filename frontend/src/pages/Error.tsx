import bedrevLogo from "/bedrevLogo.svg";
import { Link } from "react-router-dom";

interface ErrorProps {
  error: string;
}

const Error = ({ error }: ErrorProps) => {
  return (
    <div className="flex justify-center flex-col w-full h-[100vh] items-center leading-[30px] gap-8">
      <Link to={"/"} className="flex">
        <img className="w-[180px]" src={bedrevLogo} alt="companylogo" />
      </Link>
      <div className="sm:text-[18px] text-[14px]">
        <p className="text-center font-medium tracking-tight">
          Beklager, vi kunne ikke finne bedriften du leter etter :(
        </p>
        <p className="text-center font-bold tracking-wider">
          Feilmelding: {error}
        </p>
      </div>
    </div>
  );
};

export default Error;
