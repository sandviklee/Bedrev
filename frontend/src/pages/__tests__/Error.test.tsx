import { render, screen } from "@testing-library/react";
import Error from "../Error.tsx";
import { BrowserRouter as Router } from "react-router-dom";

describe("Error page", () => {
  test("renders correctly", () => {
    render(
      <Router>
        <Error error="Test error" />
      </Router>,
    );

    expect(
      screen.getByText(
        "Beklager, vi kunne ikke finne bedriften du leter etter :(",
      ),
    ).toBeInTheDocument();
    expect(screen.getByText("Feilmelding: Test error")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(
      <Router>
        <Error error="Test error" />
      </Router>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
