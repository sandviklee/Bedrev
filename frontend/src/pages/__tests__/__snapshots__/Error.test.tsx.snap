// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Error page > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center flex-col w-full h-[100vh] items-center leading-[30px] gap-8"
  >
    <a
      class="flex"
      href="/"
    >
      <img
        alt="companylogo"
        class="w-[180px]"
        src="/project2/bedrevLogo.svg"
      />
    </a>
    <div
      class="sm:text-[18px] text-[14px]"
    >
      <p
        class="text-center font-medium tracking-tight"
      >
        Beklager, vi kunne ikke finne bedriften du leter etter :(
      </p>
      <p
        class="text-center font-bold tracking-wider"
      >
        Feilmelding: Test error
      </p>
    </div>
  </div>
</DocumentFragment>
`;
